ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for time series data collection and writing said data to a file."""
+
+from datetime import datetime
+import json
+from pathlib import Path
+import time
+from typing import NamedTuple, Union
+
+from ros_cross_compile.sysroot_creator import INTERNALS_DIR
+
+
+Datum = NamedTuple('Datum', [('name', str),
+                             ('value', Union[int, float]),
+                             ('unit', str),
+                             ('timestamp', str)])
+
+
+class DataCollector:
+    """Provides an interface to collect time series data."""
+
+    def __init__(self):
+        self.data = {}
+
+    def add_datum(self, new_datum: Datum, new_datum_name: str):
```

Comment:
the Datum class already has a name, the name argument is superfluous

Reply:
Some Datum element will contain a name such as 'stage-time,' or 'stage-size.' However, for the dictionary, we just want 'stage.'

Reply:
Why do we just want 'stage'? All measurement names are distinct, there is not "namespacing". So, we may have the convention that there is "stage-time" and "stage-size", and we may understand that they are related, but the underlying database, and by proxy this collector, don't have such a grouping mechanism.